src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -316,6 +334,37 @@ private boolean isReadyToBeDeleted(Envelope envelope) {
         ).contains(envelope.getStatus());
     }
 
+    private void handleConstraintViolation(
+        String containerName,
+        String zipFilename,
+        String leaseId,
+        ConstraintViolationException exception
+    ) {
+        log.warn(
```

Comment:
If we keep this method, this warning would have to go to the `if` where we reject the envelope.